src/solvers/dg.jl

```diff
@@ -369,7 +369,11 @@ include("dgsem/dgsem.jl")
 function allocate_coefficients(mesh::AbstractMesh, equations, dg::DG, cache)
   # We must allocate a `Vector` in order to be able to `resize!` it (AMR).
   # cf. wrap_array
-  zeros(eltype(cache.elements), nvariables(equations) * nnodes(dg)^ndims(mesh) * nelements(dg, cache))
+  zeros(eltype(cache.elements), n_coefficients(mesh, equations, dg, cache))
+end
+
+@inline function n_coefficients(mesh::Union{TreeMesh, StructuredMesh, UnstructuredMesh2D, P4estMesh}, equations, dg::DG, cache)
```

Comment:
Is this function really necessary?

Reply:
We need the number of coefficients in the `SemidiscretizationCoupled` to split the `u_ode` vector to get the smaller ones for the corresponding meshes. We could also just hard-code it everywhere, but we thought it would be better to have this in one central function.

Reply:
But in `SemidiscretizationCoupled` you already have the existing other `semi`s, so couldn't you just query `ndofs(semi)` for them? Maybe I'm missing something...

Reply:
Yes, we just changed that